### PR TITLE
Improve Voice Widget

### PR DESCRIPTION
**Pull request by cribskip**
* make resizable
* fix too big logo
* improve icon

### CODE REVIEW

File: mobile/src/main/res/layout/voice_widget.xml [hunk 1]
```diff
@@ -10,12 +10,12 @@
         android:id="@+id/btn_mic"
 
         android:layout_centerHorizontal="true"
-        android:src="@drawable/menu_mic_dark"
```
Comment (mueller-ma) [inline mobile/src/main/res/layout/voice_widget.xml]: If the drawable is now unused, please remove it. As well as menu_mic_light

**Comment by cribskip** [comment]
Done after verifying those are not used anymore.

File: mobile/src/main/res/layout/voice_widget.xml [hunk 2]
```diff
@@ -10,12 +10,12 @@
...
+        android:src="@drawable/ic_mic_black_24dp"
```
Comment (mueller-ma) [inline mobile/src/main/res/layout/voice_widget.xml]: Please insert `android:tint="@color/fab_material_white"` here or any other white/grey color if you think thats suits better. But black is a bad color here.

**Comment by mueller-ma** [comment]
When I resize the widget, it looks pixelated. I dont know why, because its a vector and no bitmap. Can you try this on your device?
screenshot_ - - - - -

File: mobile/src/main/res/layout/voice_widget.xml (hunk 2 above)
Comment (cribskip) [reply mobile/src/main/res/layout/voice_widget.xml]: By adding the tint property, all goes crazy. The vector graphic get pixelatet. I'd rather keep the black icon.

**Comment by cribskip** [comment]
Works fine for me, see below.
Please also see my comment above concerning the tint property.


2017-12-21 06_33_23-android emulator - nexus_5x_api_23_5554

**Comment by mueller-ma** [comment]
With the dark mic you have a black icon on a dark background. I would like to see a white icon.

File: mobile/src/main/res/layout/voice_widget.xml [hunk 3]
```diff
@@ -10,12 +10,12 @@
         android:id="@+id/btn_mic"
```
Comment (mueller-ma) [inline mobile/src/main/res/layout/voice_widget.xml]: Please remove this empty line

**Comment by mueller-ma** [comment]
The openhab icon has a static size (32 dip height). I think you should change that to 1/3 of widget height whenever the widget is resized.

Adding this to VoiceWidget.java is a good start, but I couldnt find a way to change the layout:

```
     @Override
     public void onAppWidgetOptionsChanged(Context context, AppWidgetManager appWidgetManager,
                                           int appWidgetId, Bundle newOptions) {
         int i = (int) newOptions.get(OPTION_APPWIDGET_MIN_HEIGHT);
         int i2 = (int) newOptions.get(OPTION_APPWIDGET_MAX_HEIGHT);
         Log.e(TAG, "widget" + i + " " +i2);
     }
```

**Comment by cribskip** [comment]
I've improved as requested, I hope. Sizing is done without code thru XML settings. Please see screenshot below:
2017-12-27 20_40_01-android emulator - nexus_5x_api_23_5554

I've inverted the Mic Symbol to account for bigger sizes and added contentDescriptions to the images.

File: mobile/src/main/res/layout/voice_widget.xml [hunk 4]
```diff
@@ -1,26 +1,25 @@
...
-        android:paddingBottom="5dp"
         android:src="@mipmap/icon"
-        android:background="#C333" />
-</RelativeLayout>
\ No newline at end of file
+        android:layout_width="fill_parent"
+        android:layout_height="10dp"
+        android:layout_weight="33"
+        android:contentDescription="@string/app_name"
+        android:background="#FFF" />
```
Comment (mueller-ma) [inline mobile/src/main/res/layout/voice_widget.xml]: IMO #C333 looks better

File: mobile/src/main/res/layout/voice_widget.xml [hunk 5]
```diff
@@ -1,26 +1,28 @@
...
-        android:layout_height="match_parent"
-        android:padding="10dp"
         android:id="@+id/btn_mic"
-
-        android:layout_centerHorizontal="true"
-        android:src="@drawable/menu_mic_dark"
-        android:layout_above="@+id/btn_open_main" />
+        android:layout_width="fill_parent"
+        android:layout_height="wrap_content"
+        android:layout_weight="75"
```
Comment (mueller-ma) [inline mobile/src/main/res/layout/voice_widget.xml]: Please change this to 70. (70+30=100)

File: mobile/src/full/java/org/openhab/habdroid/core/GcmIntentService.java [hunk 6]
```diff
@@ -101,7 +101,7 @@ private void sendNotification(String msg, int notificationId) {
 
         NotificationCompat.Builder mBuilder =
                 new NotificationCompat.Builder(this)
-                        .setSmallIcon(R.drawable.ic_stat_openhab_bw)
+                        .setSmallIcon(R.mipmap.icon)
```
Comment (mueller-ma) [inline mobile/src/full/java/org/openhab/habdroid/core/GcmIntentService.java]: Seems pretty unrelated in this PR.

**Comment by mueller-ma** [comment]
I think you mixed up your branches.

**Comment by cribskip** [comment]
You are right, thanks for noticing.
I'll sort this mess out in my fork after this PR is accepted.

I'll correct my mistake tomorrow.

**Comment by cribskip** [comment]
I tried to revert this commit but could not solve this myself.
Does anybody know how to resolve this? Thanks

**Comment by digitaldan** [comment]
@cribskip what branches did you mess up?  Can you explain a little more?

**Comment by mueller-ma** [comment]
You mixed your branches in Updated notification icon to not use old ic_stat_openhab_bw.png 
So run `git reset --soft HEAD~3 && git reset`. Than add and commit the correct files, switch to a new branch `git checkout -b client-audio` and commit again. Than `git checkout master && git push --force`

**Comment by cribskip** [comment]
Ok, at least I gave it a try. This does still not look good to me.
May I recreate my fork and open a new pull request implementing the changes?

**Comment by maniac103** [comment]
You don't need a new fork. Just recreate the branch:
```
git remote add upstream git://github.com/openhab/openhab-android
git fetch upstream
git branch -m master master-old
git checkout --track -b master upstream/master
```
Then cherry pick your changes:
`git cherry-pick f75e3b4356` (etc. for your other commits)
Then try whether it works. If it does, update your repo: `git push -f origin master`

(Assuming you originally cloned your fork)

**Comment by cribskip** [comment]
@maniac103 & @mueller-ma thank you for your detailed instructions.
I've tried them to the bone but my main problem seems to be that I forked but didn't branch off.

I'd really like to start all over, rebasing my fork and braching, afterwards creating a new PR for fixing #581 , #531. What's your opinion on this?

**Comment by mueller-ma** [comment]
No problem :)
